checkov/terraform/deep_analysis_plan_graph_manager.py

```diff
@@ -0,0 +1,43 @@
+import logging
+
+from checkov.terraform.graph_builder.graph_components.blocks import TerraformBlock
+from checkov.terraform.graph_builder.local_graph import TerraformLocalGraph
+from checkov.terraform.graph_builder.graph_components.block_types import BlockType
+from checkov.common.output.report import Report
+from .plan_parser import TF_PLAN_RESOURCE_ADDRESS
+from typing import Dict
+
+
+class DeepAnalysisGraphManager:
+    def __init__(self, tf_graph: TerraformLocalGraph, tf_plan_graph: TerraformLocalGraph) -> None:
+        self.tf_graph: TerraformLocalGraph = tf_graph
+        self.tf_plan_graph: TerraformLocalGraph = tf_plan_graph
+        self._address_to_tf_vertex_map: Dict[str, TerraformBlock] = {}
+        self._address_to_tf_plan_vertex_map: Dict[str, TerraformBlock] = {}
+        self._apply_address_mapping()
+
+    def _apply_address_mapping(self) -> None:
+        for vertex in self.tf_graph.vertices:
+            if vertex.block_type == BlockType.RESOURCE:
+                self._address_to_tf_vertex_map[vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]] = vertex
```

Comment:
```suggestion
        self._address_to_tf_vertex_map = {
            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: vertex
            for vertex in self.tf_graph.vertices
            if vertex.block_type == BlockType.RESOURCE:
        }
```
I think you can do it via a dict comprehension